### ISSUE TITLE
Infinit loop using delay, force

### ISSUE THREAD

**Issue by gitmh** [biwascheme/biwascheme#58]
I just tried some examples with the new delay and force functions but they produce infinitiy loops in the browser. Here is my example:

```
(define stream-car car)

(define stream-cdr
  (lambda (stm)
    (force (cdr stm))))

(define-syntax stream-cons
  (syntax-rules ()
    ((_ head tail) (cons head (delay tail)))))

(define make-stream 
  (lambda (seed proc)
    (letrec 
      ((stream-builder
         (lambda (n)
           (stream-cons n (stream-builder (proc n))))))
      (stream-builder seed))))
  
(define stream-item
  (lambda (str k)
    (if (= k 0)
      (stream-car str)
      (stream-item (stream-cdr str) (- k 1)))))

(define stream-map
  (lambda (f str)
    (stream-cons
     (f (stream-car str))
     (stream-map f (stream-cdr str)))))

(define print-stream
  (lambda (stm n)
    (cond
      ((<= n 0) (display "...")(newline))
      (else
       (display (stream-car stm))
       (display ", ")
       (print-stream (stream-cdr stm) (- n 1))))))

(define nat_zahlen (make-stream 0 (lambda (n) (+ n 1))))
(print-stream nat_zahlen 11)

```
This example works on https://bluishcoder.co.nz/jsscheme/ and should produce:

(print-stream nat_zahlen 11)
0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, ...

**Comment by yhara** [comment]
Hmm, it looks like working fine with my browser.

- Which version are you using? Recently I've updated the implementation of `delay` and `force` with [this commit](https://github.com/biwascheme/biwascheme/commit/7b7c18748bd7e5bee0b1e2a75c813f8b23c4ca84).
- How did you define `stream-cons`, as BiwaScheme does not have `define-syntax` yet? I used the following definition.

  ```scm
  (define-macro (stream-cons head tail)
    `(cons ,head (delay ,tail)))
  ```
